include/quicly.h

```diff
@@ -210,6 +218,16 @@ typedef struct st_quicly_packet_allocator_t {
     void (*free_packet)(struct st_quicly_packet_allocator_t *self, quicly_datagram_t *packet);
 } quicly_packet_allocator_t;
 
+typedef struct st_quicly_codec_t {
+    size_t (*encrypt_packet)(struct st_quicly_codec_t *self, ptls_aead_context_t *aead, ptls_cipher_context_t *header_protection,
+                             uint8_t *dst, uint8_t *dst_payload_from, uint8_t *first_byte_at, uint64_t packet_number);
+    int (*decrypt_packet)(struct st_quicly_codec_t *self, quicly_conn_t *conn, quicly_pn_space_t *space,
+                             ptls_cipher_context_t *header_protection, ptls_aead_context_t **aead, size_t epoch,
+                             uint64_t *next_expected_pn, quicly_decoded_packet_t *packet);
+    int (*decrypt_packet_callback)(quicly_conn_t *conn, quicly_decoded_packet_t *packet, quicly_pn_space_t *space, size_t epoch, uint64_t pn,
```

Comment:
Just a suggestion, `decrypt_packet_continue` or `decrypt_packet_done` ?

Reply:
+1 for decrypt_packet_done